stdlib/InteractiveUtils/src/macros.jl

```diff
@@ -106,12 +106,27 @@ function gen_call_with_extracted_types_and_kwargs(__module__, fcn, ex0)
         end
     end
     thecall = gen_call_with_extracted_types(__module__, fcn, arg)
+    if (thecall.head === :call && thecall.args[1] === fcn)
```

Comment:
This seems a bit awkward and reliant on the exact order of things. Can we invert this by passing in the list of kwargs into the previous call or at least returning the position pointer therefrom?